packages/utils-query-operation/lib/FragmentSelectorShapes.ts

```diff
@@ -76,11 +84,21 @@ function doesShapeAcceptOperationRecurseOperation(
       return false;
     }
   }
-  if (operation.patterns && !operation.patterns
-    .every((input: Algebra.Pattern) => doesShapeAcceptOperationRecurseShape(shapeTop, shapeTop, input, options))) {
+  if (operation.expression && isExtensionFunction(operation.expression) &&
+    !doesShapeAcceptOperationRecurseShape(shapeTop, shapeTop, operation.expression, options)) {
     return false;
   }
-  return true;
+  return !(operation.patterns && !operation.patterns
+    .every((input: Algebra.Pattern) => doesShapeAcceptOperationRecurseShape(shapeTop, shapeTop, input, options)));
+}
+
+function functionIsSparql11Required(iri: string): boolean {
+  return /^https?:\/\/www\.w3\.org\//u.test(iri);
+}
```

Comment:
@rubensworks this test might be good enough (note: that is check the whole namespace though)
Interestingly enough, this works for SPARQL11 and SPARQL12 (since sparql1.2 does not introduce new required named functions). I do wonder whether newer versions of SPARQL might impose different requirements.
(Maybe that's a bridge we can cross when we get there)

Reply:
Sure, that's fine by me!

Not a big fan of the name though, as we'd have to change it for every new SPARQL version.
What about something like `isStandardSparqlFunction`?